homeassistant/components/assist_pipeline/vad.py

```diff
@@ -148,3 +148,94 @@ def _process_chunk(self, chunk: bytes) -> bool:
                 self._silence_seconds_left = self.silence_seconds
 
         return True
+
+
+@dataclass
+class VoiceActivityTimeout:
+    """Detects silence in audio until a timeout is reached."""
+
+    silence_seconds: float
+    """Seconds of silence before timeout."""
+
+    reset_seconds: float = 0.5
+    """Seconds of speech before resetting timeout."""
+
+    vad_mode: int = 3
+    """Aggressiveness in filtering out non-speech. 3 is the most aggressive."""
+
+    vad_frames: int = 480  # 30 ms
+    """Must be 10, 20, or 30 ms at 16Khz."""
+
+    _silence_seconds_left: float = 0.0
+    """Seconds left before considering voice command as stopped."""
+
+    _reset_seconds_left: float = 0.0
+    """Seconds left before resetting start/stop time counters."""
+
+    _vad: webrtcvad.Vad = None
+    _audio_buffer: bytes = field(default_factory=bytes)
```

Comment:
Similar comment as above, would a `bytesarray` be a better fit?

Reply:
Switched this over to a `bytearray` and combined the code with `VoiceActivityTimeout`